packages/nextjs/src/utils/response.ts

```diff
@@ -37,3 +37,15 @@ export const setHeader = (res: Response, name: string, val: string) => {
   res.headers.set(name, val);
   return res;
 };
+
+export const stringifyHeaders = (headers: { forEach: Headers['forEach'] }) => {
+  if (!headers) {
+    return JSON.stringify({});
+  }
+
+  const obj: Record<string, string> = {};
```

Comment:
🙃 Try this one-liner to see if it's working `JSON.stringify(Object.fromEntries(headers) || {})`.